### PR TITLE
Update asterisk on required form fields

### LINKED ISSUE #3871: Build - update form mandatory field label
Update the required label with new position of asterisk

### PR DESCRIPTION
**Pull request by bethcollins92**
## Done

- Change position of asterisk to before text
- Change colour from red to black
- Updated Vanilla version to `2.35.0`, to be inline with this [PR](https://github.com/canonical-web-and-design/vanilla-framework/pull/3935/files) when it is merged 

Fixes #3871 

## QA

- Open [demo](https://vanilla-framework-3948.demos.haus/docs/examples/patterns/forms/forms-required)
- Check the asterisk is before the text and the colour is black as seen in [design](https://app.zeplin.io/project/60e5bad7cfc6030846cacd89/screen/6114f46a421e5098c5c1e7df)
- Documentation not updated as only changed scss of current pattern 

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [x] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical-web-and-design/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical-web-and-design/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [x] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical-web-and-design/vanilla-framework/blob/master/templates/docs/whats-new.md).
- [x] Documentation side navigation should be updated with the relevant labels.

## Screenshots

Screenshot 2021-08-23 at 15 19 10

### CODE REVIEW

**Review by bartaz — APPROVED** [review]

**Comment by bartaz** [comment]
The :before doesn't work with checkboxes unfortunately:

Screenshot 2021-08-23 at 16 36 17
https://vanilla-framework-3948.demos.haus/docs/examples/patterns/forms/tick-comparison

It kinda does work with `p-radio` `p-checkbox` - it puts the asterix before the input. It is not broken, but looks weird.
But it does break 'base' checkbox/radio inputs because they use "before" to show the styled tick.

It seems we didn't really have a design for a position of asterisk with radio/checkbox.

**Comment by lyubomir-popov** [comment]
This is what I would expect to see:
image

**Comment by bartaz** [comment]
@lyubomir-popov could you have a look at this demo and percy just to confirm that's what we expect it to look like?

**Comment by lyubomir-popov** [comment]
@bartaz the dot has gone wonky

File: templates/docs/base/forms.md [hunk 1]
```diff
@@ -192,7 +192,7 @@ View example of form validation patterns
 
 ## Required
 
-By applying the class `.is-required` the attribute specifies that an input field must be filled out before submitting the form.
+By applying the class `.is-required` the attribute specifies that an input field must be filled out before submitting the form. You should also add a legend to the form to be explicit about the meaning of the asterisk.
```
Comment (lyubomir-popov) [inline templates/docs/base/forms.md:195]: I've read that active voice is preferred when writing documentation so how about 
```suggestion
Apply the class `.is-required` to mandatory input fields. You should also add a legend to the form to be explicit about the meaning of the asterisk.
```

**Review by lyubomir-popov — CHANGES_REQUESTED** [review]

File: templates/docs/base/forms.md (hunk 1 above)
Comment (bethcollins92) [reply templates/docs/base/forms.md]: Done 👍🏼

**Review by lyubomir-popov — APPROVED** [review]

**Pull request merged**